tests/sentry/workflow_engine/endpoints/validators/actions/test_msteams.py

```diff
@@ -0,0 +1,76 @@
+from unittest import mock
+
+from sentry.notifications.models.notificationaction import ActionTarget
+from sentry.testutils.cases import TestCase
+from sentry.workflow_engine.endpoints.validators.base import BaseActionValidator
+from sentry.workflow_engine.models import Action
+
+
+class TestMSTeamsActionValidator(TestCase):
+    def setUp(self) -> None:
+        super().setUp()
+        self.integration, self.org_integration = self.create_provider_integration_for(
+            provider="msteams", organization=self.organization, user=self.user, name="msteams"
+        )
+
+        self.valid_data = {
+            "type": Action.Type.MSTEAMS,
+            "config": {"targetDisplay": "cathy-sentry", "targetType": ActionTarget.SPECIFIC.value},
+            "data": {},
+            "integrationId": self.integration.id,
+        }
+
+    @mock.patch("sentry.integrations.msteams.actions.form.find_channel_id")
+    def test_validate(self, mock_check_for_channel):
+        mock_check_for_channel.return_value = "C1234567890"
+
+        validator = BaseActionValidator(
+            data=self.valid_data,
+            context={"organization": self.organization},
+        )
+
+        result = validator.is_valid()
+        assert result is True
+
+    def test_validate__missing_integration_id(self):
+        del self.valid_data["integrationId"]
+        validator = BaseActionValidator(
+            data={**self.valid_data},
+            context={"organization": self.organization},
+        )
+
+        result = validator.is_valid()
+        assert result is False
```

Comment:
it'd be cool if this also validated the error is related to integration id (i may or may not have written a test like this, and found that the validator was failing as expected, but not for the correct reasons.. and there was just a bug in code 😅)